### PR TITLE
Suggest autograd.no_grad in nn.Module.{train,eval} (and vice versa)

### LINKED ISSUE #19160: Suggest model.eval() in torch.no_grad (and vice versa)
## 📚 Documentation

`model.eval()`  and `with torch.no_grad` are both commonly used in evaluating a model.

[Confusion exists about whether setting `model.eval()` also sets no_grad](https://discuss.pytorch.org/t/does-model-eval-with-torch-set-grad-enabled-is-train-have-the-same-effect-for-grad-history/17183/5?u=ataraxy)

Would you accept a PR which suggests usage of the other, with words like:

* If evaluating a model's performance, using Module.eval() may also be useful.
* If evaluating a model's performance, using autograd.no_grad may also be useful.

### PR DESCRIPTION
**Pull request by HaleTom**
Fixes #19160: "Suggest model.eval() in torch.no_grad (and vice versa)"

### CODE REVIEW

**Comment by HaleTom** [comment]
Due to a strange [conda issue](https://stackoverflow.com/questions/43485569/installed-a-package-with-anaconda-cant-import-in-python), I can't run `make html`.

I trust that the change is small enough that I lucked out and got it right, or that the required changes are easily detectable by human eyeball.

Please inform me of any changes required!

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -1265,6 +1265,8 @@ def train(self: T, mode: bool = True) -> T:
         mode, if they are affected, e.g. :class:`Dropout`, :class:`BatchNorm`,
         etc.
 
+        Additionally, consider using :class:`~no_grad` when evaluating a model.
```
Comment (albanD) [inline torch/nn/modules/module.py:1268]: I am not sure the link here will work properly as `no_grad` is not in the `torch.nn` namespace.

File: torch/autograd/grad_mode.py [hunk 2]
```diff
@@ -62,6 +62,8 @@ class no_grad(_DecoratorContextManager):
 
     Also functions as a decorator. (Make sure to instantiate with parenthesis.)
 
+    Additionally, consider using :meth:`~torch.nn.eval` when evaluating a model.
```
Comment (albanD) [inline torch/autograd/grad_mode.py:65]: I don't think the link is right here either, it's missing a `Module` no?

Also I would add a bit more details here, what do you think?
Additionally, when working with `nn.Module`, you should consider using `nn.Module.eval` to ensure that `nn.Module` behave properly for evaluation (deactivate dropout, set batchnorm to use saved statistics, etc).

File: torch/nn/modules/module.py [hunk 3]
```diff
@@ -1287,6 +1289,8 @@ def eval(self: T) -> T:
 
         This is equivalent with :meth:`self.train(False) <torch.nn.Module.train>`.
 
+        Additionally, consider using :class:`~no_grad` when evaluating a model.
```
Comment (albanD) [inline torch/nn/modules/module.py:1292]: Like above, I would add a bit more detail here like: `, to de-activate the autograd.`

**Review by albanD — COMMENTED** [review]
Thanks for the PR,
Added some feedback in comments.

File: torch/nn/modules/module.py (hunk 1 above)
Comment (HaleTom) [reply torch/nn/modules/module.py:1268]: Do you have any ideas about what it should be?

Comment (albanD) [reply torch/nn/modules/module.py:1268]: By building locally, doing ```:class:`torch.no_grad` ``` does generate the right link.
